### PR TITLE
abstract gadgets resource classes

### PR DESCRIPTION
**Pull request by lehmannk**
this commit adds abstract gadgets resource and interface classes
for confapi plugins to implement

### CODE REVIEW

File: src/main/java/de/aservo/atlassian/confapi/rest/api/GadgetsResource.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                    @ApiResponse(responseCode = "default", content = @Content(schema = @Schema(implementation = ErrorCollection.class))),
+            }
+    )
+    Response getGadgets();
+
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(
+            tags = { ConfAPI.LICENSES },
```
Comment (pathob) [inline src/main/java/de/aservo/atlassian/confapi/rest/api/GadgetsResource.java:40]: `ConfAPI.GADGETS`

File: src/main/java/de/aservo/atlassian/confapi/rest/api/GadgetsResource.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            }
+    )
+    Response setGadgets(
+            @NotNull final GadgetsBean gadgetsBean);
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(
+            tags = { ConfAPI.LICENSES },
```
Comment (pathob) [inline src/main/java/de/aservo/atlassian/confapi/rest/api/GadgetsResource.java:55]: `ConfAPI.GADGETS`

File: src/main/java/de/aservo/atlassian/confapi/rest/api/GadgetsResource.java (hunk 1 above)
Comment (lehmannk) [reply src/main/java/de/aservo/atlassian/confapi/rest/api/GadgetsResource.java:40]: oh ja, sorry, fix kommt sofort.

File: src/main/java/de/aservo/atlassian/confapi/rest/api/GadgetsResource.java (hunk 2 above)
Comment (lehmannk) [reply src/main/java/de/aservo/atlassian/confapi/rest/api/GadgetsResource.java:55]: oh ja, sorry, fix kommt sofort.

**Pull request merged**